subgraphs/notional-finance/src/mappings/notional.ts

```diff
@@ -168,6 +170,13 @@ export function handleLendBorrowTrade(event: LendBorrowTrade): void {
     if (!activeMarkets.includes(allMarkets.activeMarkets[k])) {
       const m = getOrCreateMarket(event, allMarkets.activeMarkets[k]);
 
+      // calculate supply side revenue before we set market to inactive
+      if (m.lendRates && m.lendRates!.length != 0) {
+        const avgInterestRate = avg(m.lendRates!);
+        updateFinancials(event, m.id, avgInterestRate);
```

Comment:
why do we want to update financial snapshots only when the market is going to be closed?

Reply:
Thought this was sufficient for us. Was also considering errors associated with cumulative values dropping (if there are withdrawals). I went ahead and moved `updateFinancials` to the `createDeposit` method.

Reply:
I would be concerned with financial snapshots not being created. I don't imagine markets being closed every day right?

Reply:
How would cumulative values drop? Those should be strictly increasing fields.

I like the methodology of calling updater functions each time a user interacts with the protocol and changes state. Deposit, withdrawal, repay, etc.

Reply:
> I like the methodology of calling updater functions each time a user interacts with the protocol and changes state. Deposit, withdrawal, repay, etc.

Agree, I do the same. In the previous case, since we were doing it at market close, it was moved here. But it's now part of the `deposit` events so we should be good to go on this one.

Reply:
> How would cumulative values drop? Those should be strictly increasing fields.
This was because of an implementation issue on my end previously. I thought I pushed the fix but looks like I didn't, will update now.